lib/Genesis/Kit/Provider/Github.pm

```diff
@@ -0,0 +1,434 @@
+package Genesis::Kit::Provider::Github;
+use strict;
+use warnings;
+
+use base 'Genesis::Kit::Provider';
+use Genesis;
+use Genesis::Helpers;
+
+use constant {
+	DEFAULT_DOMAIN => 'github.com',
+	DEFAULT_LABEL  => "Custom Github-based Kit Provider",
+	DEFAULT_TLS    => 'yes'
+};
+
+### Class Methods {{{
+
+# init - creates a new provider on repo init or change {{{
+sub init {
+	my ($class, %opts) = @_;
+	my $label = $opts{"kit-provider-name"} || DEFAULT_LABEL;
+
+	bail("$label kit provider requires specifying the organization using the --kit-provider-org option")
+		unless $opts{"kit-provider-org"};
+	
+	$opts{"kit-provider-tls"} ||= DEFAULT_TLS;
+	bail("Github kit provider option --kit-provider-tls only accepts: no, yes, and skip")
+		unless $opts{"kit-provider-tls"} =~ /^(no|yes|skip)$/;
+	$class->new(
+		type         => 'github',
+		label         => $label,
+		domain       => $opts{'kit-provider-domain'},
+		organization => $opts{'kit-provider-org'},
+		tls          => $opts{'kit-provider-tls'}
+	);
+}
+
+# }}}
+# new - create a default genesis-community kit provider {{{
+sub new {
+	my ($class, %config) = @_;
+	my $credentials;
+	if ($ENV{GITHUB_USER} && $ENV{GITHUB_AUTH_TOKEN}) {
+		$credentials = "$ENV{GITHUB_USER}:$ENV{GITHUB_AUTH_TOKEN}";
+	}
+	bless({
+		domain          => $config{domain} || DEFAULT_DOMAIN,
+		organization    => $config{organization},
+		credentials     => $credentials,
+		label           => $config{label} || DEFAULT_LABEL,
+		tls             => $config{tls} || DEFAULT_TLS,
+	}, $class);
+}
+
+# }}}
+# opts -  list of options supported by init method {{{
+sub opts {
+	qw/
+		kit-provider-name=s
+		kit-provider-domain=s
+		kit-provider-org=s
+		kit-provider-tls=s
+		/;
+}
+
+# }}}
+# opts_help - specifies the new/update options understood by this provider {{{
+sub opts_help {
+	<<EOF
+  Kit Provider `github`:
+
+    This is a generic kit provider type for kits backed by Github or Github
+    Enterprise, allowing you to specify the domain url and organization to
+    target your specific provider location.  It supports the following options:
```

Comment:
Maybe note that this only spans a single organization.  I got the impression from our in-review conversation on Friday that I could specify multiple orgs in a single provider (which justified the pomp and circumstance around all this future flexibility 😁 )